Added new settings to logcollector module, ignore and restrict

jotacarma90 opened the pull request:
## Description

This PR is to add the description of the two new settings in the logcollector reference. Added description, example cases and allowed value.
Issue related: https://github.com/wazuh/wazuh/issues/5628
PR related: https://github.com/wazuh/wazuh/pull/14782

## Checks
- [x] It compiles without warnings.
- [x] Spelling and grammar. 
- [x] Used impersonal speech. 
- [x] Used uppercase only on nouns. 
- [ ] Updated the `redirect.js` script if necessary (check [this guide](https://github.com/wazuh/wazuh-documentation/blob/master/NEW_RELEASE.md)).

branchnetconsulting commented:
I'd recommend clarifying what happens if multiple `<restrict_log>` lines or multiple `<ignore_log>` lines are used in the same `<localfile>` section.  Will the regexes for the matching tag lines just be concatenated or will the separate matching tag lines be interpreted as being in an OR relationship?  

It would also be good to state that if `<restrict_log>` and `<ignore_log>` are both used in the same `<localfile>` section, that `<ignore_log>` takes precedence when the ignore and restrict pattern both match the same line (assuming that is the behavior).

chemamartinez commented on source/user-manual/reference/ossec-conf/localfile.rst:
```diff
@@ -34,6 +34,8 @@ Options
 - `exclude`_
 - `reconnect_time`_
 - `multiline_regex`_
+- `ignore_log`_
```
```suggestion
- `ignore`_
```
Please apply this suggestion to the whole PR.

chemamartinez commented on source/user-manual/reference/ossec-conf/localfile.rst:
```diff
@@ -34,6 +34,8 @@ Options
 - `reconnect_time`_
 - `multiline_regex`_
+- `ignore_log`_
+- `restrict_log`_
```
```suggestion
- `restrict`_
```

jotacarma90 commented:
Hi @branchnetconsulting 
Indeed we should clarify some points, I have updated the documentation and implemented some new features:
- Choose the type of regex expression (osregex, osmatch, pcre2. The default is pcre2).
- Possibility to use ignore and restrict multiple times (OR for ignore, AND for restrict).
- In case any ignore expression and any restrict expression are identical, no log will be processed. I have clarified it also in the docu.

Thank you very much!

chemamartinez commented on source/user-manual/reference/ossec-conf/localfile.rst:
```diff
@@ -595,6 +597,96 @@ For example, we may want to read a Python Traceback output as one single log, re
+ignore
+^^^^^^^^^^^^^^^
+
+This allows you to configure a pcre2 expression to ignore specific log lines. Multiple expressions can be configured by repeating the ignore label. The log will be ignored as soon as one of the expressions matches, OR behavior.
```
- Better using impersonal sentences (avoid using `This allows you`).
- pcre2 is uppercase
- Let's simplify the description
```suggestion
Specify a regular expression to ignore log lines or command outputs when matching. Whether several `ignore` labels are defined, entries are ignored when matching any of the specified ones.
```

chemamartinez commented on source/user-manual/reference/ossec-conf/localfile.rst:
```diff
@@ -595,6 +597,96 @@ For example, we may want to read a Python Traceback output as one single log, re
++--------------------+---------------------------------------------------------------+
+| **Allowed values** | Any `regex <regex.html#regex-os-regex-syntax>`_,              |
+|                    | `sregex <regex.html#sregex-os-match-syntax>`_ or              |
+|                    | `pcre2 <regex.html#pcre2-syntax>`_ expression.                |
```
```suggestion
|                    | `PCRE2 <regex.html#pcre2-syntax>`_ expression.                |
```

chemamartinez commented on source/user-manual/reference/ossec-conf/localfile.rst:
```diff
@@ -595,6 +597,96 @@ For example, we may want to read a Python Traceback output as one single log, re
+|                    | `pcre2 <regex.html#pcre2-syntax>`_ expression.                |
++--------------------+---------------------------------------------------------------+
+
+The attribute type is optional.
```
```suggestion
Use the `type` attribute to define the regular expression type. By default, PCRE2 is applied.
```

chemamartinez commented on source/user-manual/reference/ossec-conf/localfile.rst:
```diff
@@ -595,6 +597,96 @@ For example, we may want to read a Python Traceback output as one single log, re
+  </localfile>
+
+.. note::
+  For formats that group multiple lines, the entire group will be counted as a single log to check if it should be restricted.
```
Better use present than future.
```suggestion
  For formats that group multiple lines, the entire group is treated as a single log when evaluating the regex.
```

chemamartinez commented on source/user-manual/reference/ossec-conf/localfile.rst:
```diff
@@ -595,6 +597,96 @@ For example, we may want to read a Python Traceback output as one single log, re
+  For formats that group multiple lines, the entire group will be counted as a single log to check if it should be restricted.
+
+.. note::
+  The ignore label is checked before the restrict label, so if you set identical expressions for both settings, no log will be processed.
```
```suggestion
  Whether the same log entry matches an ignore and also a restrict configured for the same `localfile`, the entry is discarded. In other words, the `ignore` has precedence over `restrict`. Said that, if the same expression is defined in both `ignore` and `restrict`, no log will be processed for that `localfile`.
```

chemamartinez commented on source/user-manual/reference/ossec-conf/localfile.rst:
```diff
@@ -595,6 +597,96 @@ For example, we may want to read a Python Traceback output as one single log, re
+  The ignore label is checked before the restrict label, so if you set identical expressions for both settings, no log will be processed.
+
+.. note::
+  On Windows, the ``eventchannel`` format already provides a way to restrict logs through queries, so this setting ``does not apply`` to this format.
```
```suggestion
  The `eventchannel` format already provides a way to filter logs through queries. Therefore, `ignore` and `restrict` settings don't apply to this format.
```

chemamartinez commented on source/user-manual/reference/ossec-conf/localfile.rst:
```diff
@@ -595,6 +597,96 @@ For example, we may want to read a Python Traceback output as one single log, re
+      <ignore type="osregex">type=CONFIG_\.+</ignore>
+  </localfile>
+
+.. note::
```
These three notes are duplicated, it is cleaner if we add it only once at the end of the `restrict` section.

chemamartinez commented on source/user-manual/reference/ossec-conf/localfile.rst:
```diff
@@ -595,6 +597,96 @@ For example, we may want to read a Python Traceback output as one single log, re
+restrict
+^^^^^^^^^^^^^^^
+
+This allows you to configure a pcre2 expression to restrict specific log lines, avoiding to generate alerts for those logs that do not match. Multiple expressions can be configured by repeating the restrict label. The log will be processed only if all expressions match.
```
```suggestion
Specify a regular expression to restrict processed log lines or command outputs. Whether several `restrict` labels are defined, entries are processed when matching all of them.
```

chemamartinez commented on source/user-manual/reference/ossec-conf/localfile.rst:
```diff
@@ -595,6 +597,96 @@ For example, we may want to read a Python Traceback output as one single log, re
++--------------------+---------------------------------------------------------------+
+| **Allowed values** | Any `regex <regex.html#regex-os-regex-syntax>`_,              |
+|                    | `sregex <regex.html#sregex-os-match-syntax>`_ or              |
+|                    | `pcre2 <regex.html#pcre2-syntax>`_ expression.                |
```
```suggestion
|                    | `PCRE2 <regex.html#pcre2-syntax>`_ expression.                |
```

chemamartinez commented on source/user-manual/reference/ossec-conf/localfile.rst:
```diff
@@ -595,6 +597,96 @@ For example, we may want to read a Python Traceback output as one single log, re
+|                    | `pcre2 <regex.html#pcre2-syntax>`_ expression.                |
++--------------------+---------------------------------------------------------------+
+
+The attribute type is optional.
```
```suggestion
Use the `type` attribute to define the regular expression type. By default, PCRE2 is applied.
```

chemamartinez commented on source/user-manual/reference/ossec-conf/localfile.rst:
```diff
@@ -595,6 +597,96 @@ For example, we may want to read a Python Traceback output as one single log, re
+
+The attribute type is optional.
+
++-------------+-----------------------------------------+-------------+---------------+
```
Use this table format:
<img width="719" alt="image" src="https://user-images.githubusercontent.com/29475387/193076056-4eea389a-0ce8-45ee-a7e9-501c7059f462.png">

chemamartinez commented on source/user-manual/reference/ossec-conf/localfile.rst:
```diff
@@ -595,6 +597,96 @@ For example, we may want to read a Python Traceback output as one single log, re
+  <localfile>
+      <log_format>syslog</log_format>
+      <location>/custom/file/path</location>
+      <restrict type="osregex">username_\.+</restrict>
```
Change the type of this `restrict` to show that types can be different in the same `localfile`.

chemamartinez commented on source/user-manual/reference/ossec-conf/localfile.rst:
```diff
@@ -595,6 +597,96 @@ For example, we may want to read a Python Traceback output as one single log, re
+  <localfile>
+      <log_format>audit</log_format>
+      <location>/var/log/audit/audit.log</location>
+      <ignore type="osregex">type=\.+_CHANGE</ignore>
```
Change the type of this `ignore` to show that types can be different in the same `localfile`.

chemamartinez reviewed (approved):
LGTM!

Arcapas reviewed (approved):
LGTM
